node/core/provisioner/src/lib.rs

```diff
@@ -550,46 +550,57 @@ mod tests {
 	mod select_availability_bitfields {
 		use super::super::*;
 		use super::{default_bitvec, occupied_core};
+		use futures::executor::block_on;
 		use lazy_static::lazy_static;
-		use polkadot_primitives::v1::{SigningContext, ValidatorIndex, ValidatorPair};
-		use sp_core::crypto::Pair;
-		use std::sync::Mutex;
+		use polkadot_primitives::v1::{SigningContext, ValidatorIndex, ValidatorId};
+		use sp_core::traits::CryptoStorePtr;
+		use sp_application_crypto::AppKey;
+		use std::sync::{Arc, Mutex};
+		use sc_keystore::LocalKeystore;
 
 		lazy_static! {
 			// we can use a normal mutex here, not a futures-aware one, because we don't use any futures-based
 			// concurrency when accessing this. The risk of contention is that multiple tests are run in parallel,
 			// in independent threads, in which case a standard mutex suffices.
-			static ref VALIDATORS: Mutex<HashMap<ValidatorIndex, ValidatorPair>> = Mutex::new(HashMap::new());
+			static ref VALIDATORS: Mutex<HashMap<ValidatorIndex, ValidatorId>> = Mutex::new(HashMap::new());
 		}
 
-		fn signed_bitfield(
+		async fn signed_bitfield(
+			keystore: &CryptoStorePtr,
 			field: CoreAvailability,
 			validator_idx: ValidatorIndex,
 		) -> SignedAvailabilityBitfield {
 			let mut lock = VALIDATORS.lock().unwrap();
-			let validator = lock
+			let public = keystore.sr25519_generate_new(ValidatorId::ID, None)
+				.await
+				.expect("generated sr25519 key");
+			lock
 				.entry(validator_idx)
-				.or_insert_with(|| ValidatorPair::generate().0);
+				.or_insert(public.into());
 			SignedAvailabilityBitfield::sign(
+				&keystore,
 				field.into(),
 				&<SigningContext<Hash>>::default(),
 				validator_idx,
-				validator,
-			)
+				&public.into(),
+			).await.expect("Should be signed")
 		}
 
 		#[test]
 		fn not_more_than_one_per_validator() {
+			let keystore_path = tempfile::tempdir().expect("Creates keystore path");
+			let keystore : CryptoStorePtr = Arc::new(LocalKeystore::open(keystore_path.path(), None)
```

Comment:
Likely a dump question: Wouldn't a in memory keystore suffice here?

Reply:
Good question. However, generating keys without seeds would skip adding those keys to `additional` (cache) and go straight to writing them on the filesystem. Without configuring the keystore to use the filesystem, adding any key would generate the key and then lose sight of it. This is why i opted for configuring the keystore to write on the filesystem.

Reply:
> Without configuring the keystore to use the filesystem, adding any key would generate the key and then lose sight of it

Could you elaborate? We are storing the pubkeys in the `VALIDATORS` hashmap though, why do we need a seed? Are you saying that if we replace `LocalKeystore::open` with `LocalKeystore::in_memory` the tests will fail?

If so, this warrants a comment in code.

Reply:
I am not sure `VALIDATORS` here is needed anymore but didn't want to remove it. However, to answer your question, and according to substrate's [keystore impl](https://github.com/paritytech/substrate/blob/596f377d49fede088e0c32791fb72f1343003b0d/client/keystore/src/lib.rs#L175), generating a key without a seed would trigger this function which requires that the keystore's path is configured rather than using the in-memory storage.

I will add a comment about this.

Reply:
Comment added. `VALIDATORS` removed.

Reply:
Thanks for the clarification. 
It's still can be avoided maybe by e.g. using dump seeds. But it's only used in tests anyway, so probably not worth time spent.

Reply:
> I am not sure VALIDATORS here is needed anymore but didn't want to remove it.

I think it's still needed (we're expecting the same key for the given `ValidatorIndex`), cc @coriolinus